beacon/goclient/goclient.go

```diff
@@ -141,18 +142,26 @@ type goClient struct {
 	registrationMu       sync.Mutex
 	registrationLastSlot phase0.Slot
 	registrationCache    map[phase0.BLSPubKey]*api.VersionedSignedValidatorRegistration
+	commonTimeout        time.Duration
+	maxTimeout           time.Duration
 }
 
 // New init new client and go-client instance
 func New(logger *zap.Logger, opt beaconprotocol.Options, operatorID spectypes.OperatorID, slotTickerProvider slotticker.Provider) (beaconprotocol.BeaconNode, error) {
-	logger.Info("consensus client: connecting", fields.Address(opt.BeaconNodeAddr), fields.Network(string(opt.Network.BeaconNetwork)))
+	const (
+		commonTimeout = time.Second * 5
+		maxTimeout    = time.Second * 120
+	)
 
-	httpClient, err := http.New(opt.Context,
+	logger.Info("consensus client: connecting", fields.Address(opt.BeaconNodeAddr), fields.Network(string(opt.Network.BeaconNetwork)))
+	dialCtx, cancel := context.WithTimeout(opt.Context, commonTimeout)
+	defer cancel()
```

Comment:
The context gets canceled at the end of `New`, but it's passed to `httpClient`, which is used in `goClient` for beacon calls. Is this intended behavior?

Reply:
@nkryuchkov only `dialCtx` is canceled and it's passed only to `eth2clienthttp.New` which doesn't keep it for later (like we do in `goClient` which is wrong by the way 😅)

Reply:
@moshe-blox As I understand, `eth2clienthttp.New` keeps it for later, not as a struct field, but instead it spawns a goroutine that closes connection on context done:

https://github.com/attestantio/go-eth2-client/blob/master/http/service.go#L138-L143

Reply:
@nkryuchkov woah, good catch 🎯

see https://github.com/bloxapp/ssv/pull/1293#issuecomment-1916536035